### PR TITLE
Add InitFlow and move session states to context

### PR DESCRIPTION
**Pull request by mirromutth**
Motivation:

Session states are now dispersed across `MySqlSimpleConnection` and `ConnectionContext`, make these states to be managed in a single object.

Modification:

- Add `InitFlow` for initialize handshake/login and session states
- Add `setSessionVariable` and `setSessionVariables` to `QueryFlow`
- Consolidate session variable statements into `StringUtils`
- Move session states to `ConnectionConext`, e.g. `PrepareCache`, `lockWaitTimeout`, etc.
  - All session states should be reset/recreate after reconnect or redirect, see also #89 , #228 
  - Remove `ConnectionState`, because all states should be used by `ConnectionContext`

Result:

Refactored initialization and management of session states

### CODE REVIEW

**Comment by jchrys** [comment]
I just had a medical check-up; I'll review PRs tomorrow.

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/InitFlow.java [hunk 1]
```diff
@@ -0,0 +1,747 @@
...
+     * @param client                the {@link Client} to exchange messages with.
+     * @param sslMode               the {@link SslMode} defines SSL capability and behavior.
+     * @param database              the database that will be connected.
+     * @param user                  the user that will be login.
+     * @param password              the password of the {@code user}.
+     * @param compressionAlgorithms the list of compression algorithms.
+     * @param zstdCompressionLevel  the zstd compression level.
+     * @return a {@link Flux} that indicates the initialization is done, or an error if the initialization failed.
+     */
+    static Flux<Void> initHandshake(Client client, SslMode sslMode, String database, String user,
```
Comment (jchrys) [inline r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/InitFlow.java]: maybe Mono?

**Review by jchrys — APPROVED** [review]
LGTM

**Comment by jchrys** [comment]
Great change 👍

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/InitFlow.java (hunk 1 above)
Comment (mirromutth) [reply r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/InitFlow.java]: Yes, you are right, `Flux<Void>` should not be used as a return type

**Pull request merged**